Sources/AsyncHTTPClient/HTTPClient.swift

```diff
@@ -312,6 +342,25 @@ public class HTTPClient {
         case createNew
     }
 
+    /// Specifies how the library will treat event loop passed by the user.
+    public struct EventLoopPreference {
+        enum Preference {
+            /// Event Loop will be selected by the library.
+            case indifferent
+            /// Library will try to use provided event loop if possible.
+            case prefers(EventLoop)
+        }
+
+        var preference: Preference
+
+        /// Event Loop will be selected by the library.
+        public static let indifferent = EventLoopPreference(preference: .indifferent)
+        /// Library will try to use provided event loop if possible.
+        public static func prefers(_ eventLoop: EventLoop) -> EventLoopPreference {
+            return EventLoopPreference(preference: .prefers(eventLoop))
```

Comment:
the parameter shouldn't have a name here, "preference" is duplicated. Should be `EventLoopPreference(.prefers(eventLoop))`

Reply:
good idea, fixed!